engine/consensus/approvals/interface.go

```diff
@@ -0,0 +1,17 @@
+package approvals
+
+import (
+	"github.com/onflow/flow-go/engine/consensus/approvals/tracker"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type AssignmentCollector interface {
+	BlockID() flow.Identifier
+	ResultID() flow.Identifier
+
+	ProcessIncorporatedResult(incorporatedResult *flow.IncorporatedResult) error
+	ProcessApproval(approval *flow.ResultApproval) error
+
```

Comment:
We could add a method 

```suggestion

	ProcessingStatus() ProcessingStatus

	// ChangeProcessingStatus changes the AssignmentCollector's internal processing
	// status. The operation is implemented as an atomic compare-and-swap, i.e. the
	// state transition is only executed if AssignmentCollector's internal state is
	// equal to `expectedValue`. The boolean return indicates whether the state
	// was updated.
	ChangeProcessingStatus(expectedValue, newValue ProcessingStatus) (bool, error)
	
```
with 
```golang
type ProcessingStatus int
const (
	Caching ProcessingStatus = iota
	Verifying
	Orphaned
)
```